module/optee/mbx/test/mod_optee_mbx_unit_test.c

```diff
@@ -27,10 +30,369 @@ void tearDown(void)
+
+    /* Fail to bind at round 1 */
+    element_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_MBX, FAKE_SCMI_CHANNEL_DEVICE_IDX_OSPM_0);
+    fwk_id_is_type_ExpectAndReturn(element_id, FWK_ID_TYPE_MODULE, false);
+    fwk_id_get_element_idx_ExpectAndReturn(element_id, FAKE_SCMI_CHANNEL_DEVICE_IDX_OSPM_0);
+    fwk_module_bind_ExpectAndReturn(
+        device_ctx->config->driver_id,
+        device_ctx->config->driver_api_id,
+        &device_ctx->shmem_api,
+        FWK_E_PARAM);
+    status = mbx_bind(element_id, 1);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /*  Succeed to bind at round 1 */
+    element_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_MBX, FAKE_SCMI_CHANNEL_DEVICE_IDX_OSPM_0);
+    fwk_id_is_type_ExpectAndReturn(element_id, FWK_ID_TYPE_MODULE, false);
+    fwk_id_get_element_idx_ExpectAndReturn(element_id, FAKE_SCMI_CHANNEL_DEVICE_IDX_OSPM_0);
+    fwk_module_bind_ExpectAndReturn(
+        device_ctx->config->driver_id,
+        device_ctx->config->driver_api_id,
+        &device_ctx->shmem_api,
+        FWK_SUCCESS);
+    status = mbx_bind(element_id, 1);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(device_ctx->shmem_id.value, device_ctx->config->driver_id.value);
+}
+
+/*!
+ * \brief test module's start function.
+ *
+ *  \details Test all cases for start function.
+ */
+void test_optee_mbx_start(void)
+{
+    int status;
+    fwk_id_t module_id;
+
+    /* Start always succeed */
+    module_id = FWK_ID_MODULE(FWK_MODULE_IDX_OPTEE_MBX);
+    status = mbx_start(module_id);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+}
+
+/*!
+ * \brief test optee_mbx_get_devices_count function.
+ *
+ *  \details Test all cases for optee_mbx_get_devices_count function.
+ */
+void test_optee_mbx_get_devices_count(void)
+{
+    int status;
+
+    status = optee_mbx_get_devices_count();
+    TEST_ASSERT_EQUAL(FAKE_SCMI_CHANNEL_DEVICE_IDX_COUNT, status);
+}
+
+/*!
+ * \brief test optee_mbx_get_device function.
+ *
+ *  \details Test all cases for optee_mbx_get_device function.
+ */
+void test_optee_mbx_get_device(void)
+{
+    fwk_id_t device_id, returned_id;
+    unsigned int id;
+
+    /* Fail to get out of range index */
+    id = FAKE_SCMI_CHANNEL_DEVICE_IDX_COUNT;
+    device_id = optee_mbx_get_device(id);
+    returned_id = (fwk_id_t)FWK_ID_NONE_INIT;
+    TEST_ASSERT_EQUAL(returned_id.value, device_id.value);
+
+    /* Succeed to get a device */
+    id = FAKE_SCMI_CHANNEL_DEVICE_IDX_OSPM_0;
+    device_id = optee_mbx_get_device(id);
+    returned_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_MBX, id);
+    TEST_ASSERT_EQUAL(returned_id.value, device_id.value);
+}
+
+static int optee_signal_message(fwk_id_t channel_id)
+{
+    return FWK_SUCCESS;
+}
+
+static struct mod_optee_smt_driver_input_api optee_smt_input_api = {
+    .signal_message = optee_signal_message,
+};
+
+/*!
+ * \brief test optee mbx with static memory.
+ *
+ *  \details Test all cases for ignaling a message to a static memory.
```

Comment:
How could this test test all cases?
It's only one scenario, no?

Reply:
looks like I forgot to add some tests